src/generators/markdowndocgenerator/toolbargenerator.tsx

```diff
@@ -0,0 +1,64 @@
+import { MarkdownDocGeneratorOptionsManager } from './optionsmanager';
+
+import * as React from 'react';
+
+interface NotebookGeneratorToolbarProps {}
+
+interface NotebookGeneratorToolbarState {
+  numbering: boolean;
+}
+
+export function markdownDocGeneratorToolbar(
+  options: MarkdownDocGeneratorOptionsManager
+) {
+  // Render the toolbar
+  return class extends React.Component<
```

Comment:
Maybe this is a React-ism that I am unfamiliar with. Why are you returning a class, rather than an instance of a class?

Reply:
For this issue, I tried to return a React instance instead of returning a React class, but this made the whole TOC extension crashed.

Then I tried returning JSX elements, but I found that the states of the toolbar are no longer tracked, therefore, every time user click something in the toolbar, the toolbar doesn't know which part should be updated and need to re-render the whole toolbar, and this can be inefficient. (especially if the cell tags menu is open)

Do you feel like it is better for us to change it to return JSX elements or continuing using this HoC? Thank you.

Reply:
I still don't understand why this requires a higher order component to work. However, I'm comfortable with moving forwards with this as it is, and we can revisit whether it needs to be a HoC later.

Reply:
Thank you!